src/pip/_internal/index/collector.py

```diff
@@ -424,7 +424,10 @@ def _handle_get_page_fail(
 ) -> None:
     if meth is None:
         meth = logger.debug
-    meth("Could not fetch URL %s: %s - skipping", link, reason)
+    url = str(link)
+    if logger.getEffectiveLevel() > logging.DEBUG:
+        url = link.show_url
+    meth("Could not fetch URL %s: %s - skipping", url, reason)
```

Comment:
I don't think warnings/errors should present partial URLs, since its possible that the user has multiple indexes and needs to know which one the error is related to, without adding additional verbosity.